lib/transformers/TermTransformer.ts

```diff
@@ -67,13 +75,27 @@ export class TermTransformer implements ITermTransformer {
     const dataType = lit.datatype.value;
     const superTypeDict: GeneralSuperTypeDict = getSuperTypeDict(dataType, this.superTypeProvider);
 
+    // The order of checking matters! Check most specific types first!
     if (TypeURL.XSD_STRING in superTypeDict) {
       return new E.StringLiteral(lit.value, dataType);
     }
-    if (DT.RDF_LANG_STRING in superTypeDict) {
+    if (TypeURL.RDF_LANG_STRING in superTypeDict) {
       return new E.LangStringLiteral(lit.value, lit.language);
     }
-    if (DT.XSD_DATE_TIME in superTypeDict) {
+    if (TypeURL.XSD_YEAR_MONTH_DURATION in superTypeDict) {
+      return new E.YearMonthDurationLiteral(yearMonthDurationParser(lit.value), lit.value, dataType);
+    }
+    if (TypeURL.XSD_DAY_TIME_DURATION in superTypeDict) {
+      return new E.DayTimeDurationLiteral(dayTimeDurationParser(lit.value), lit.value, dataType);
+    }
+    // TODO: maybe the special notations need to be checked so we don't Years in a DayTimeDuration? ...
```

Comment:
> maybe the special notations need to be checked so we don't have a Years field in a DayTimeDuration?

I think this is done? But yes. This should be checked since it would be an invalid literal Form!